src/arfsdao.ts

```diff
@@ -137,27 +137,32 @@ export class ArFSDAOAnonymous extends ArFSDAOType {
 	}
 
 	async getPublicDrive(driveId: string): Promise<ArFSPublicDrive> {
+		const driveBuilder = await this.getPublicDriveMetaData(driveId);
+		return driveBuilder.build();
+	}
+
+	async getPublicDriveMetaData(driveId: string): Promise<ArFSPublicDriveBuilder> {
 		const gqlQuery = buildQuery([
 			{ name: 'Drive-Id', value: driveId },
 			{ name: 'Entity-Type', value: 'drive' },
 			{ name: 'Drive-Privacy', value: 'public' }
 		]);
-
 		const response = await this.arweave.api.post(graphQLURL, gqlQuery);
-
 		const { data } = response.data;
 		const { transactions } = data;
 		const { edges } = transactions;
-
 		if (!edges.length) {
 			throw new Error(`Public drive with Drive ID ${driveId} not found!`);
 		}
-
 		const driveBuilder = new ArFSPublicDriveBuilder(driveId, this.arweave);
-		return driveBuilder.build();
+		return driveBuilder;
 	}
 
 	async getPublicFolder(folderId: string): Promise<ArFSPublicFolder> {
+		const folderBuilder = await this.getPublicFolderMetaData(folderId);
+		return await folderBuilder.build();
+	}
```

Comment:
Needs whitespace between functions (although I'm not convinced these functions should have been broken up in the first place).